app/Colors.php

```diff
@@ -341,4 +345,39 @@ public static function getContrast($hexcolor)
 		$contrastRatio = 1.9; // higher number = more black color
 		return hexdec($hexcolor) > 0xffffff / $contrastRatio ? 'black' : 'white';
 	}
+
+	/**
+	 * Update field color code and generate stylesheet file.
+	 *
+	 * @param int    $fieldId
+	 * @param string $color
+	 *
+	 * @return bool
+	 */
+	public static function updateFieldColor($fieldId, $color): bool
+	{
+		$fieldHasColor = (bool) (new \App\Db\Query())->select(['color'])->from('vtiger_field')->where(['fieldid' => $fieldid])->scalar();
+		$updateFieldColorResult = (bool) Db::getInstance()->createCommand()->update('vtiger_field', ['color' => $color], ['fieldid' => $fieldId])->execute();
+		static::generate('field');
+		if ($updateFieldColorResult || !$fieldHasColor) {
+			return true;
+		}
+		return false;
```

Comment:
one return
one query

Reply:
That was made because I would to know if will update a record which is exists because I will get 0